### DISCUSSION TITLE
GECKO 3.0 on Human cell lines - Questions with parameters and proteomics integration

### DISCUSSION THREAD

**Discussion by h-escoffier** [SysBioChalmers/GECKO#397] [Q&A]
Dear all, 

I am interested in using your GECKO 3.0 method to simulate various human cell lines. After reviewing your method and conducting several trials, I have a few questions to ensure I can effectively run your tool:

1. In Step 4, you mention entering several variables, including _Ptot_, _f_, and _sigma_. In the light_ecModel example, specific values are provided (based on the average for NCI60). Would it be possible to clarify how obtain these values in the required units?

3. In the example, you include the parameter `c_UptakeExp = 0.641339301; %[mmol/gDw h]/Average across NCI60 cell lines`. Since this parameter is not referenced in your article, what it represents?

4. In Step 54, regarding the integration of proteomic data, you mention converting the data into the specified units (milligram per gram dry cell weight). Should I develop my own method for this conversion, or is there a tool available for this purpose?

I'm very interested in using your tool to the best of my ability and look forward to hearing from you. 
Thank you very much for your time, and I wish you happy holidays!

**Comment by Yu-sysbio** [comment]
Hi, thanks for your interest in the GECKO 3.0 toolbox! Regarding your questions, below are my answers:
1. The product of the three variables is used as a constraint on the protein pool usage reaction. If one of them is unknown, then the product is unknown. In most cases sigma is unknown and thus the product of the three variables is unknown. Thus I do not really obtain exact values for all the three variables. Instead I prefer to follow Steps 40 and 41 to estimate the protein pool usage. 
2. This parameter represents the measured uptake rate of the nutrient e.g. glucose. To calculate this value for the cell line of interest, you may culture the cells in bioreactors and measure the changes in the nutrient concentration and cell weight over time.
3. There is no tool for the conversion in the toolbox as we do not know which unit the users would provide. Thus you have to do the conversion yourself.

Just let me know if I did not explain clearly and maybe some other developers correct me.
Looking forward to further discussion.

**Comment by manas-kohli** [comment]
Hi Yu,

I wanted to clarify a few more things regarding point number 3. I originally built some enzyme constrained models using getSubsetEcModel from the light_ecModel from the tutorial. I'd like to add some specific proteomics data to this now. There is a function called constrainEnzymes in the ecModels repo but this function seems a bit old so I'm wondering if it's still compatible with models built via the GECKO3 framework because in the light_ecModel tutorial it's mentioned that proteomics integration isn't possible. When I tried to build the full_ecModel for a human-GEM, I reached as far as step 2 in the full tutorial but when I now try to make a model subset, I consistently run into the issue that the maximum possible biomass production is 0 (I opened uptake reactions and the same works for the light_ecModel when I try to get a subset). Do you have any advice on how to proceed. Happy to share code and models as required for this. Thanks in advance!

**Reply by edkerk** [reply]
- GECKO3 models are not compatible with earlier functions, so `constrainEnzymes` will indeed not work.
- light_ecModels do not support proteomics integration. The reason for this is that the _light_ model formulation does not have enzyme exchange reactions that can be constrained by proteomics measurements.
- Regarding step 2, I'll look at #405 in detail
